exercises/practice/zebra-puzzle/.meta/example.php

```diff
@@ -0,0 +1,223 @@
+<?php
+
+/*
+ * By adding type hints and enabling strict type checking, code can become
+ * easier to read, self-documenting and reduce the number of potential bugs.
+ * By default, type declarations are non-strict, which means they will attempt
+ * to change the original type to match the type specified by the
+ * type-declaration.
+ *
+ * In other words, if you pass a string to a function requiring a float,
+ * it will attempt to convert the string value to a float.
+ *
+ * To enable strict mode, a single declare directive must be placed at the top
+ * of the file.
+ * This means that the strictness of typing is configured on a per-file basis.
+ * This directive not only affects the type declarations of parameters, but also
+ * a function's return type.
+ *
+ * For more info review the Concept on strict type checking in the PHP track
+ * <link>.
+ *
+ * To disable strict typing, comment out the directive below.
+ */
+
+declare(strict_types=1);
+
+class ZebraPuzzle
+{
+    private $waterDrinker = '';
+    private $zebraOwner = '';
+
+    private const FIRST = 1;
+    private const MIDDLE = 3;
+
+    private $red = 0;
+    private $green = 0;
+    private $ivory = 0;
+    private $yellow = 0;
+    private $blue = 0;
+
+    private $englishman = 0;
+    private $spaniard = 0;
+    private $ukrainian = 0;
+    private $japanese = 0;
+    private $norwegian = 0;
+
+    private $coffee = 0;
+    private $tea = 0;
+    private $milk = 0;
+    private $orangeJuice = 0;
+    private $water = 0;
+
+    private $oldGold = 0;
+    private $kools = 0;
+    private $chesterfields = 0;
+    private $luckyStrike = 0;
+    private $parliaments = 0;
+
+    private $dog = 0;
+    private $snails = 0;
+    private $fox = 0;
+    private $horse = 0;
+    private $zebra = 0;
+
+    private $nationalityNames = [];
+
+    private $possiblePermutations;
+
+    public function __construct()
+    {
+        $this->possiblePermutations = $this->permuteValues([1, 2, 3, 4, 5]);
+        $this->solve();
+    }
+
+    public function waterDrinker(): string
+    {
+        return $this->waterDrinker;
+    }
+
+    public function zebraOwner(): string
+    {
+        return $this->zebraOwner;
+    }
+
+    private function permuteValues(array $array): array
+    {
+        $result = [];
+
+        $length = count($array);
+
+        if ($length === 0) {
+            return [[]];
+        }
+
+        foreach ($array as $index => $Value) {
```

Comment:
Note to self: How could this pass the CI?

```suggestion
        foreach ($array as $index => $value) {
```

Reply:
I have updated the value variable.